### PR TITLE
Require at least one tensor to be marked dynamic with --dynamic-batch-only

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99634
* #99617
* __->__ #99620

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: benchmarks/dynamo/torchbench.py [hunk 1]
```diff
@@ -282,6 +282,10 @@ def load_model(
         if self.args.accuracy and model_name in MAX_BATCH_SIZE_FOR_ACCURACY_CHECK:
             batch_size = min(batch_size, MAX_BATCH_SIZE_FOR_ACCURACY_CHECK[model_name])
 
+        # See https://github.com/pytorch/benchmark/issues/1560
+        if model_name == "speech_transformer":
+            batch_size = 10
```
Comment (voznesenskym) [inline benchmarks/dynamo/torchbench.py:287]: por que 10

File: benchmarks/dynamo/torchbench.py [hunk 2]
```diff
@@ -317,6 +321,10 @@ def load_model(
         # the right example_inputs
         if model_name == "yolov3":
             example_inputs = (torch.rand(batch_size, 3, 384, 512).to(device),)
+        # See https://github.com/pytorch/benchmark/issues/1561
+        if model_name == "maml_omniglot":
+            batch_size = 5
```
Comment (voznesenskym) [inline benchmarks/dynamo/torchbench.py:326]: and why 5

**Review by voznesenskym — APPROVED** [review]

File: benchmarks/dynamo/torchbench.py (hunk 1 above)
Comment (ezyang) [reply benchmarks/dynamo/torchbench.py:287]: That's empirically what the batch size torchbench pops out is

**Pull request closed without merging**